pandas/core/strings.py

```diff
@@ -74,10 +74,12 @@ def cat_core(list_of_columns: List, sep: str):
     """
     if sep == "":
         # no need to interleave sep if it is empty
-        return np.sum(list_of_columns, axis=0)
+        arr_of_cols = np.asarray(list_of_columns, dtype=object)
```

Comment:
What is this cleaning up?

Reply:
this is for when numpy re-institutes the make-object-explicit thing from #30035.

Reply:
I think should hold off on these changes then until that comes back

Reply:
they reverted it specifically to give downstream time to update without breaking CIs

Reply:
great this is good